### PR TITLE
feat(azure-devops): select work item type

### PR DESCRIPTION
**Pull request by scefali**
This PR adds a required form field for the work item type for Azure DevOps which determines what kind of work item we create. We use [this endpoint ](https://docs.microsoft.com/en-us/rest/api/azure/devops/wit/work%20item%20type%20categories/list?view=azure-devops-rest-4.1)to query the work item categories for a particular project. If the user changes the project, the form updates (`updatesForm=True`) and we load the categories for the new project. We also automatically save the previous choice from the user so the next time someone tries to make a new work item, the old value will be pre-selected.

Here's what the form looks like:
Screen Shot 2020-08-26 at 9 54 20 AM

### CODE REVIEW

**Review by mgaeta — APPROVED** [review]
:+1:

File: src/sentry/integrations/vsts/issues.py [hunk 1]
```diff
@@ -75,7 +110,17 @@ def get_create_issue_config(self, group, **kwargs):
...
+                "updatesForm": True,
+            },
+            {
+                "name": "work_item_type",
+                "required": True,
+                "type": "choice",
+                "choices": work_item_choices,
+                "defaultValue": default_work_item,
+                "label": _("Work Item Type"),
+                "placeholder": _("MyWorkItem"),
```
Comment (MeredithAnya) [inline src/sentry/integrations/vsts/issues.py:122]: I feel like an example type would be better like "Bug"

File: src/sentry/integrations/vsts/client.py [hunk 2]
```diff
@@ -149,11 +159,8 @@ def get_work_item_states(self, instance, project):
             api_preview=True,
         )
 
-    def get_work_item_types(self, instance, process_id):
-        return self.get(
-            VstsApiPath.work_item_types.format(instance=instance, process_id=process_id),
-            api_preview=True,
-        )
+    def get_work_item_categories(self, instance, project):
```
Comment (MeredithAnya) [inline src/sentry/integrations/vsts/client.py:162]: I thought we didn't want the categories? we did want the types?

Comment (scefali) [reply src/sentry/integrations/vsts/client.py:162]: @MeredithAnya We use the category API and then pull the work item types from the response

**Review by MeredithAnya — APPROVED** [review]

**Pull request merged**